plugins/woocommerce-blocks/assets/js/atomic/utils/blocks-registration-manager/template-change-detector.ts

```diff
@@ -96,28 +134,15 @@ export class TemplateChangeDetector implements TemplateChangeDetectorSubject {
 		return templateId as string;
 	}
 
-	/**
-	 * Checks if the current template or page has changed and notifies subscribers.
-	 *
-	 * If the current template ID has changed and is not undefined (which means that it is not a page, post or template), it notifies all subscribers.
-	 */
-	public checkIfTemplateHasChangedAndNotifySubscribers(): void {
-		this.previousTemplateId = this.currentTemplateId;
-
-		const templateId = this.getCurrentTemplateIdFromUrl(
-			window.location.href
-		);
-
-		this.currentTemplateId = this.parseTemplateId( templateId );
-
-		const hasChangedTemplate =
-			this.previousTemplateId !== this.currentTemplateId;
-		const hasTemplateId = Boolean( this.currentTemplateId );
+	private getPostOrPageIdFromUrl( url: string ): string | undefined {
+		const path = getPath( url );
+		const isPostOrPage = path ? path.includes( 'post.php' ) : false;
+		let postId;
 
-		if ( ! hasChangedTemplate || ! hasTemplateId ) {
-			return;
+		if ( isPostOrPage ) {
+			postId = getQueryArg( url, 'post' );
```

Comment:
In the context of the Site Editor the query arg for page IDs is `postId` so we would likely need to cover that case too